### PR TITLE
mp3cat init at 0.5

### PR DESCRIPTION
**Pull request by OmnipotentEntity**
Adding package mp3cat, a command line mp3 utility which will concatenate
multiple mp3 files, and keep only the audio frames, discarding headers
and so on.

###### Motivation for this change

It's a reasonably useful single use utility.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/audio/mp3cat/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  version = "0.5";
+
+  src = fetchFromGitHub {
+    owner = "tomclegg";
+    repo = "mp3cat";
+    rev = version;
+    sha256 = "0n6hjg2wgd06m561zc3ib5w2m3pwpf74njv2b2w4sqqh5md2ymfr";
+  };
+
+  configurePhase = ''export PREFIX=$out'';
```
Comment (worldofpeace) [inline pkgs/tools/audio/mp3cat/default.nix]:
```suggestion
  makeFlags = [
    "PREFIX=${placeholder ''out''}"
    "CC:=$(CC)"
  ];
```

File: pkgs/tools/audio/mp3cat/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchFromGitHub {
+    owner = "tomclegg";
+    repo = "mp3cat";
+    rev = version;
+    sha256 = "0n6hjg2wgd06m561zc3ib5w2m3pwpf74njv2b2w4sqqh5md2ymfr";
+  };
+
+  makeFlags = [
+    "PREFIX=${placeholder ''out''}"
+    "CC:=$(CC)"
```
Comment (worldofpeace) [inline pkgs/tools/audio/mp3cat/default.nix]: I've checked and I don't see `$(CC)` being used in the `Makefile` (it just uses `cc`) so we can remove this.

File: pkgs/tools/audio/mp3cat/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  name = "mp3cat-${version}";
```
Comment (c0bw3b) [inline pkgs/tools/audio/mp3cat/default.nix]:
```suggestion
  pname = "mp3cat";
```

File: pkgs/tools/audio/mp3cat/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  version = "0.5";
+
+  src = fetchFromGitHub {
+    owner = "tomclegg";
+    repo = "mp3cat";
```
Comment (c0bw3b) [inline pkgs/tools/audio/mp3cat/default.nix]:
```suggestion
    repo = pname;
```

File: pkgs/tools/audio/mp3cat/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  installTargets = [
+    "install_bin"
+  ];
+
+  meta = with stdenv.lib; {
+    description = "A command line program which concatenates MP3 files";
+    longDescription = ''
+      A command line program which concatenates MP3 files, mp3cat
+      only outputs MP3 frames with valid headers, even if there is extra gargage
```
Comment (c0bw3b) [inline pkgs/tools/audio/mp3cat/default.nix]:
```suggestion
      only outputs MP3 frames with valid headers, even if there is extra garbage
```

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -23951,4 +23951,5 @@ in
 
   wasmtime = callPackage ../development/interpreters/wasmtime {};
 
+  mp3cat = callPackage ../tools/audio/mp3cat {};
```
Comment (c0bw3b) [inline pkgs/top-level/all-packages.nix]: This should be in the `### TOOLS` section, just before mp3fs

Comment (c0bw3b) [inline pkgs/top-level/all-packages.nix]: This should be in the `### TOOLS` section, just before mp3fs

**Comment by worldofpeace** [comment]
yay @c0bw3b reviews are always helpful :sparkles:

**Comment by OmnipotentEntity** [comment]
Thanks so much for the suggestions.  That typo is especially embarrassing.

**Comment by c0bw3b** [comment]
No worries.
You last pushed commit seems identical.

**Comment by OmnipotentEntity** [comment]
Somehow managed to do everything but actually add the files.  Good catch.

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build mp3cat

**Review by c0bw3b — APPROVED** [review]

**Pull request merged**